cli/lsp/diagnostics.rs

```diff
@@ -58,6 +59,15 @@ async fn publish_diagnostics(
   let maybe_changes = collection.take_changes();
   if let Some(diagnostic_changes) = maybe_changes {
     for specifier in diagnostic_changes {
+      // Skip diagnostics for JSON(c) and markdown files.
```

Comment:
Please do suggest if there's a better place to skip diagnostics for these files.

Reply:
I would rather do the opposite (only process diagnostics for file types we know we produce diagnostics for), and use the `MediaType` enum to determine that versus "guessing" at the extension. If we need to add enum variants to deal with it, it is better to have a "authoritative" place to determine the type of file.

Reply:
I moved the logic to the individual `generate_*_diagnostic` functions.  Let me know your thoughts on new changes.